Sources/PackageGraph/RepositoryPackageContainerProvider.swift

```diff
@@ -219,8 +225,8 @@ public class LocalPackageContainer: BasePackageContainer, CustomStringConvertibl
         toolsVersionLoader: ToolsVersionLoaderProtocol,
         currentToolsVersion: ToolsVersion,
         fs: FileSystem = localFileSystem
-    ) {
-        assert(URL.scheme(identifier.path) == nil, "unexpected scheme \(URL.scheme(identifier.path)!) in \(identifier.path)")
+    ) throws {
+        try AbsolutePath.validate(path: identifier.path)
```

Comment:
This is the wrong place to fix this issue. We should treat these as a manifest parse error and not allow loading manifest that do this. We do some amount of checking for URLs when loading the manifest, this check can go in there: https://github.com/apple/swift-package-manager/blob/master/Sources/PackageLoading/PackageDescription4Loader.swift#L283

Reply:
Should this method then gets error handling added which propagates it to the caller?

Reply:
Ah, I see. Yeah, that makes much more sense. I'll change it.